### PR TITLE
Apply retry filter to ndkit conversions

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
Reduce the number of pod retries when there are failures in downloading images. Instead retry
inside the pod. This implements suggestion from https://github.com/kubevirt/containerized-data-importer/issues/2561.
Also fix some linter warnings in same package.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
BugFix: use retry filter in inline streaming conversion with qcow2 http endpoints.
```

### CODE REVIEW

File: pkg/image/nbdkit.go [hunk 1]
```diff
@@ -107,15 +103,16 @@ func NewNbdkitCurl(nbdkitPidFile, user, password, certDir, socket string, extraH
...
-	return &Nbdkit{
+	n := &Nbdkit{
 		NbdPidFile: nbdkitPidFile,
 		plugin:     NbdkitCurlPlugin,
 		nbdkitArgs: args,
 		pluginArgs: pluginArgs,
 		redactArgs: redactArgs,
 		Socket:     socket,
 	}
+	n.AddFilter(NbdkitRetryFilter)
```
Comment (akalenyu) [inline pkg/image/nbdkit.go:114]: maybe add a note that this should be added last?

File: pkg/image/qemu_test.go [hunk 2]
```diff
@@ -301,7 +301,7 @@ var _ = Describe("Create blank image", func() {
 		size := convertQuantityToQemuSize(quantity)
 		replaceExecFunction(mockExecFunction("", "", nil, "create", "-f", "raw", "image", size), func() {
 			err = CreateBlankImage("image", quantity, false)
-			Expect(err).NotTo(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("no such file or directory"))
```
Comment (akalenyu) [inline pkg/image/qemu_test.go:304]: hmm are we fine with this error happening? shouldn't we pass a tempdir to CreateBlankImage and keep test the same?

**Comment by akalenyu** [comment]
Looking at the func test failure, maybe it is retrying on a 404? unless the file host returns something else

File: pkg/image/qemu_test.go (hunk 2 above)
Comment (awels) [reply pkg/image/qemu_test.go:304]: Well that error is happening because the file we pass in is not real, and since we were ignoring the error before...

Comment (akalenyu) [reply pkg/image/qemu_test.go:304]: I think it may be worth to create a temp file and avoid the err, I think we do this in other qemu tests already

**Comment by awels** [comment]
Yeah, we were retrying at the container level on a 404 to start with. Now it is in the filter. I fixed the broken test by disabling exponential retries on the filter level. So now the filter retry will fail in sufficient time for the test to see a container retry and restore the file to the http server.

File: pkg/image/qemu_test.go (hunk 2 above)
Comment (akalenyu) [reply pkg/image/qemu_test.go:304]: Let me know if you want to do that

**Pull request merged**